doc/tutorial/08_tls_enforced.md

```diff
@@ -0,0 +1,91 @@
+## Preface 
+
+This tutorial will take you through the connection path from client to `pgagroal` to `postgres` server when TLS is enforced.
+
+## Setup
+
+To enforce tls along the whole path, we first need to create X509 certicates for client->pgagroal and pgagroal->postgres seperately.
+
+For the purpose of this tutorial we will create self-signed certificates and assume only server side authentication.
+
+### Creating Certificates
+
+We will create self-signed certificate for the server, valid for 365 days, use the following OpenSSL command, replacing `dbhost.yourdomain.com` with the server's host name, here `localhost`:
+
+```
+openssl req -new -x509 -days 365 -nodes -text -out pgagroal.crt \
+  -keyout pgagroal.key -subj "/CN=dbhost.yourdomain.com"
+```
+
+for client to pgagroal side authentication and 
+
+```
+openssl req -new -x509 -days 365 -nodes -text -out postgres.crt \
+  -keyout postgres.key -subj "/CN=dbhost.yourdomain.com"
+```
+
+for pgagroal to postgres side authentication.
+
+then do -
+
+```
+chmod og-rwx pgagroal.key
+chmod og-rwx postgres.key
+```
+
+because the server will reject the file if its permissions are more liberal than this. For more details on how to create your server private key and certificate, refer to the OpenSSL documentation.
+
+### Configuration 
+
+Modify the configuration files of postgres and pgagroal.
+
+Add the following lines in `postgresql.conf` (Generally can be found in `/etc/postgresql/<version_number>/main` directory)
+
+```
+...
+ssl = on
+ssl_cert_file = </path/to/postgres.crt>
+ssl_key_file = </path/to/postgres.key>
+...
+```
+
+and make the contents of `pg_hba.conf` -
+
+```
+hostssl all all all md5
+```
+
+here we are choosing md5 for authenticating the requested user and database against postgres catalog
```

Comment:
md5 -> scram-sha-256

Reply:
The scram-sha-256 is giving `channal binding check failed` errors because we are switching ssl sessions between actual client and postgres server.

Is there some way to bypass it?

Or we will have to present same certificates on both sides (`c_ssl == s_ssl`)?